### PR TITLE
Add environment variables to template rendering

### PR DESCRIPTION
**Pull request by DaspawnW**
The credentials endpoint in Vault don't return for example the name of the database or the host information about the database.
To have one file with all db related connection information it would be good to also include host information or database name information. Therefore the following change allows to render environment variables provided to the Pod, e.g.:

```yaml
env:
  - name: HOST
    value: localhost:5432
  - name: DBNAME
    value: postgresql
```

```properties
spring.datasource.url: jdbc:postgresql://{{ .HOST }}/{{ .DBNAME }}
spring.datasource.username: {{ .Username }}
spring.datasource.password: {{ .Password }}
```

Output:

```properties
spring.datasource.url: jdbc:postgresql://localhost:5432/postgresql
spring.datasource.username: User
spring.datasource.password: VERYSECUREPASSWORD
```

### CODE REVIEW

File: cmd/main.go [hunk 1]
```diff
@@ -60,6 +61,21 @@ func cleanUp(leasePath, tokenPath string) {
 	}
 }
 
+func contactWithEnvVariables(creds vault.Credentials) map[string]string {
```
Comment (Joseph-Irving) [inline cmd/main.go]: I'd probably change this to something like `appendEnvVars` or `templateWithEnvVars` to make it more obvious what it's doing

**Comment by Joseph-Irving** [comment]
Hey thanks for this PR! looks like a good useful change, I left one nitpick

File: cmd/main.go (hunk 1 above)
Comment (DaspawnW) [reply cmd/main.go]: Hi @Joseph-Irving,
Fixed your suggestion :-)

**Comment by DaspawnW** [comment]
I'd be very happy if you'd take over the merge. Thank you for the note :-)

**Review by Joseph-Irving — APPROVED** [review]

**Pull request merged**